comp/process/runner/runner.go

```diff
@@ -0,0 +1,60 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package runner
+
+import (
+	"testing"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/process/submitter"
+)
+
+// runner implements the Component.
+type runner struct {
+	checks    []Check
+	submitter submitter.Component
+}
+
+type dependencies struct {
+	fx.In
+
+	Checks    []Check `group:"check"`
+	Submitter submitter.Component
+}
+
+func newRunner(deps dependencies) (Component, error) {
```

Comment:
Note: if the function never fails, you can remove the `error`.

Reply:
Agreed, this will change with the actual implementation.